operations/poplar.go

```diff
@@ -143,38 +148,41 @@ func poplarReport() cli.Command {
 			certFile := c.String(certFileFlagName)
 			caFile := c.String(caFileFlagName)
 			keyFile := c.String(keyFileFlagName)
+			dryRun := c.Bool(dryRunFlagName)
 
 			report, err := poplar.LoadReport(fileName)
 			if err != nil {
 				return errors.WithStack(err)
 			}
 
-			rpcOpts := []grpc.DialOption{
-				grpc.WithUnaryInterceptor(aviation.MakeRetryUnaryClientInterceptor(10)),
-				grpc.WithStreamInterceptor(aviation.MakeRetryStreamClientInterceptor(10)),
-			}
-			if isInsecure {
-				rpcOpts = append(rpcOpts, grpc.WithInsecure())
-			} else {
-				var tlsConf *tls.Config
-				tlsConf, err = getTLSConfig(caFile, certFile, keyFile)
-				if err != nil {
-					return errors.WithStack(err)
+			if !dryRun {
+				rpcOpts := []grpc.DialOption{
+					grpc.WithUnaryInterceptor(aviation.MakeRetryUnaryClientInterceptor(10)),
+					grpc.WithStreamInterceptor(aviation.MakeRetryStreamClientInterceptor(10)),
+				}
+				if isInsecure {
+					rpcOpts = append(rpcOpts, grpc.WithInsecure())
+				} else {
+					var tlsConf *tls.Config
+					tlsConf, err = getTLSConfig(caFile, certFile, keyFile)
+					if err != nil {
+						return errors.WithStack(err)
+					}
+
+					rpcOpts = append(rpcOpts, grpc.WithTransportCredentials(credentials.NewTLS(tlsConf)))
 				}
 
-				rpcOpts = append(rpcOpts, grpc.WithTransportCredentials(credentials.NewTLS(tlsConf)))
-			}
-
-			ctx, cancel := context.WithCancel(context.Background())
-			defer cancel()
+				ctx, cancel := context.WithCancel(context.Background())
+				defer cancel()
 
-			conn, err := grpc.DialContext(ctx, addr, rpcOpts...)
-			if err != nil {
-				return errors.WithStack(err)
-			}
+				conn, err := grpc.DialContext(ctx, addr, rpcOpts...)
+				if err != nil {
+					return errors.WithStack(err)
+				}
 
-			if err := rpc.UploadReport(ctx, report, conn); err != nil {
-				return errors.WithStack(err)
+				if err := rpc.UploadReport(ctx, report, conn); err != nil {
```

Comment:
We should run through the upload code in a dry-run ish way.

Reply:
So add a dry run flag to UploadReport?